crates/sui-indexer/tests/integration_tests.rs

```diff
@@ -387,6 +430,156 @@ pub mod pg_integration_test {
+            show_type: true,
+            show_owner: true,
+            show_previous_transaction: true,
+            show_display: true,
+            show_content: true,
+            show_bcs: true,
+            show_storage_rebate: true,
+        };
+        let resp = indexer_rpc_client
+            .get_object_with_options(source_object_id, Some(show_all_content.clone()))
+            .await
+            .unwrap();
+        let initial_full_obj_data = resp.object()?;
+        let tx_response = sign_and_transfer_object(
+            &test_cluster,
+            &indexer_rpc_client,
+            &test_cluster.get_address_0(),
+            &test_cluster.get_address_1(),
+            source_object_id,
+            None,
+        )
+        .await?;
+        wait_until_transaction_synced(&store, tx_response.digest.base58_encode().as_str()).await;
+
+        let response = indexer_rpc_client
+            .get_object_with_options(source_object_id, Some(show_all_content.clone()))
+            .await?;
+        let post_transfer_full_obj_data = response.object()?;
+        let object_required_fields = SuiObjectData {
+            type_: None,
+            owner: None,
+            previous_transaction: None,
+            storage_rebate: None,
+            display: None,
+            content: None,
+            bcs: None,
+            ..post_transfer_full_obj_data.clone()
+        };
+        let show_some_content = SuiObjectDataOptions::new();
+        let futures = vec![
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(SuiObjectDataOptions::bcs_lossless()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(SuiObjectDataOptions::full_content()),
+            ),
+            indexer_rpc_client
+                .get_object_with_options(source_object_id, Some(show_some_content.clone())),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_content()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_owner()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_type()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_display()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_bcs()),
+            ),
+            indexer_rpc_client.get_object_with_options(
+                source_object_id,
+                Some(show_some_content.clone().with_previous_transaction()),
```

Comment:
happy to take any thoughts you have on rewriting this neater

Reply:
it is a bit verbose here, but it's okay